index.js

```diff
@@ -118,7 +118,11 @@ const stop = Symbol('stop');
 
 class HTTPError extends Error {
 	constructor(response) {
-		super(response.statusText);
+		super(
+			response.statusText ||
+			(response.status && String(response.status)) ||
```

Comment:
I think we should handle the case where `response.status` is `0`. I realize it is non-standard, but I found a few search results of real-world cases where people receive `0` as a status code.

Example: CloudMailin uses status code 0 for "Unreachable"

https://docs.cloudmailin.com/receiving_email/http_status_codes/

One way to solve this is to use [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator).

```js
super(response.statusText || String(response.status ?? 'Unknown response error'));
```

Reply:
Sounds like a good idea, thanks. I'll also add a test for it.